docs/assignment2.md

```diff
@@ -1 +1,409 @@
+The job of connecting both sides is done by the object gameState of type *Game*. gameState holds the *Map* that all players interact with, all Players and the timeStamp of when the game was started.
+With the figural and textual description of a state our system can have, it should be clear why each class exists, and their relations to one another.
+
+## 
+
+## State machine diagrams
+
+Author(s): `Irene Garcia-Fortea Garcia`, `Eduardo Lira` & `Theresa Schantz`
+
+<h5>State-Machine Diagram for Getting Inputs for Player</h5>
+
+> State machine diagram of Game (Diagram 3) **DESCRIPTIVE**
+
+[![State-Machine-Diagram-Page-1.png](https://i.postimg.cc/tTHWn9NV/State-Machine-Diagram-Page-1-1.png)](https://postimg.cc/gwfHpY8M) 
+
+The diagram above shows the State Machine Diagram of the event that of executing a command. After starting the process the system waits for a user input. After the input has been received, the function *getCommand()* executes. 
+
+In the next step we move to the state that the command has been received. The command needs to be checked, whether it is a valid command that the system accepts in *validateCommand()*. 
+
+If the command i not valid, an error message is printed and the program returns to the idle state, waiting for a command. 
+
+If the command is valid, the command needs to be executed (“*Execute Command*”). There, changes will be made to the Item or Player (Described in the *Execute Command* figure). After the changes have been made, the system goes back to the idle state, waiting for a new command to process, unless the user types “quit”. Then, the game terminates. 
+
+The *Execute Command* state works as the following: After checking the command for validity, it will be matched matching it with each of the following existing commands: move, help or look. 
+
+If the command is move: the player will be moved into the wanted direction (input by the player). 
+
+If the command is help: a help message will be printed to the user.
+
+If the command is look: the description of the *Area*, in that the *Player* is currently in, will be printed to the user. 
+
+<h5>State-Machine Diagram for Player Movement</h5>
+
+> Figure representing the state machine diagram of the class Map (Diagram 4) **DESCRIPTIVE**
+
+[![State-Machine-Diagram-Page-2.png](https://i.postimg.cc/FHgLtyV2/State-Machine-Diagram-Page-2-Page-2.png)](https://postimg.cc/PLZ2PpXG)
```

Comment:
Events for the decision node are not specified. The name of the states correspond errouneously to event conditions.